src/base/http/requestparser.cpp

```diff
@@ -109,9 +121,8 @@ RequestParser::ParseResult RequestParser::doParse(const QByteArray &data)
         return {ParseStatus::OK, m_request, headerLength};
     if (m_request.method == HEADER_REQUEST_METHOD_POST)
     {
-        bool ok = false;
-        const int contentLength = m_request.headers[HEADER_CONTENT_LENGTH].toInt(&ok);
-        if (!ok || (contentLength < 0))
+        const int contentLength = parseContentLength(m_request.headers[HEADER_CONTENT_LENGTH]);
```

Comment:
IMO, it would be more correct to explicitly check whether such a header exists. Otherwise you can miss some kind of "bad request", no?

Reply:
From the linked issue, I'm under the impression that some app would omit this header when its length is `0`, meaning that qbt should treat missing as `length = 0` by default.

Reply:
>And other cases of bad values will be represented by negative values from now on.

I thought about broken requests containing an empty "Content-Length" header (i.e. name without any value). Shouldn't it be treated as an obvious "Bad request"? Or such a case is handled in different place of parsing logic?

Reply:
You are right. I pushed another update.